docs/technical-documentation/flysystem.md

```diff
@@ -0,0 +1,50 @@
+
+The Fedora Flysystem adapter does not use Gemini to map the relationship between Drupal URIs and Fedora URIs, so they are indexed separately using the "files_in_fedora" Context which triggers the "Index Fedora File in Gemini" and "Delete Fedora File in Gemini" actions as appropriate.
+
+### Configuration
+
+The fedora file system is configured in the site's `settings.php` file. An example configuration can be seen in the claw-playbook webserver role's [drupal tasks](https://github.com/Islandora-Devops/claw-playbook/blob/master/roles/internal/webserver-app/tasks/drupal.yml#L12-L19):
+```
+$settings['flysystem'] = [
+  'fedora' => [
+    'driver' => 'fedora',
+    'config' => [
+      'root' => 'http://localhost:8080/fcrepo/rest/',
+    ],
+  ],
+];
+```
+The configuration array's top-level key is the name of the Drupal stream wrapper, which also serves as the filesystem prefix. Any Drupal file path using "fedora://" will use this Flysystem adapter. Drupal will translate this prefix to the site's domain plus "\_flystem/fedora/". For example, using the default configuration provided by the claw-playbook, a file stored at "fedora://test.tif" will persist to Fedora with the URI "http://localhost:8080/fcrepo/rest/test.tif" and will be accessible from the Drupal URL "http://localhost:8000/_flysystem/fedora/test.tif". The 'driver' value 'fedora' corresponds to [the plugin's machine name](https://github.com/Islandora-CLAW/islandora/blob/8.x-1.x/src/Flysystem/Fedora.php#L21). The 'config' section contains all the adapter-specific configurations. In this case, the only thing configured for the site is the Fedora REST end-point. (Change this value to match your own Fedora's location, if needed.) The JWT is configured separately.
+
+Other examples of Drupal Flysystem configurations can be seen in [the module's README](http://cgit.drupalcode.org/flysystem/plain/README.md?h=8.x-1.x).
+
+Islandora is configured to have all Media use the Fedora file system by default in the islandora_core_feature. For example, the [field storage uri_scheme setting for field_media_image](https://github.com/Islandora-CLAW/islandora/blob/8.x-1.x/modules/islandora_core_feature/config/install/field.storage.media.field_media_image.yml#L17) (and the other media types) is "fedora". This can also be viewed in the UI on the field's "Field settings" page; e.g. `http://localhost:8000/admin/structure/media/manage/image/fields/media.image.field_media_image/storage`, look for "Upload destination" and see that "Flysystem: fedora" is selected.
+
```

Comment:
@seth-shaw-unlv If "Islandora is configured to have all Media use the Fedora file system by default in the islandora_core_feature.", why did we have to modify an action to get thumbnails into Fedora in #1037 ?

Reply:
Duh, since thumbnails are derivatives.... sorry, ignore dummy's question.